src/blocks/server.blocks/middleware/__error/middleware__error.node.js

```diff
@@ -53,23 +52,39 @@ modules.define('middleware__error', ['config', 'logger', 'util'], function (prov
 
     /**
      * Log error message and set statusCode to response
+     * If the error code does not have its own template
+     * show 500 page and logging this error in console
      * @param err - {Error} error
-     * @param req - {Object} request object
      * @param res - {Object} response object
      */
-    function preparation(err, req, res) {
-        var code = err.code || 500,
-            terr = terror.ensureError(err);
+    function preparation(err, res) {
+        var supportedCodes = [404, 500],
+            code = supportedCodes.indexOf(err.code) !== -1 ? err.code : 500;
+
+        if (!err.stack) {
+            // Module intel waiting field stack in error object
+            err.stack = '';
+        }
+
+        if (err.app) {
+            try {
```

Comment:
I'd make it like this:
```js
var message = err.message || {};

try {
    message = JSON.parse(message);
} catch(e) {}

_.extend(/* ... */);
/* ... */
```
because
1. You'll need all the info even if message wasn't parsed.
2. The less code is inside `try catch` the higher performance you'll get.